packages-ui/roosterjs-react/lib/ribbon/component/buttons/cellShade.ts

```diff
@@ -0,0 +1,22 @@
+import RibbonButton from '../../type/RibbonButton';
+import { applyCellShading } from 'roosterjs-editor-api';
+import { BackgroundColorKeys, CellShadeButtonStringKey } from '../../type/RibbonButtonStringKeys';
+import {
+    BackgroundColorDropDownItems,
+    BackgroundColors,
+    getColorPickerDropDown,
+} from './colorPicker';
+
+/**
+ * @internal
+ * "Cell Shade" button on the format ribbon
+ */
+export const cellShade: RibbonButton<CellShadeButtonStringKey> = {
```

Comment:
this button only works for table, right? So we should add isDisabled() function to check if focus is in table. You can add a "isInTable" property in ElementBasedFormatState, and set it in getElementBasedFormatState().

If that is too much change, you can do this in a separate PR, and do not add this button in getButtons() for now.